tests/test_server.py

```diff
@@ -762,3 +764,205 @@ async def test_method_implementation(self, header_param: str, body_param: str) -
+    resource_action = data.ResourceAction(
+        environment=env.id,
+        version=0,
+        resource_version_ids=[f"std::File[agent1,path=/etc/motd],v={0}"],
+        action_id=earliest_action_id,
+        action=const.ResourceAction.deploy,
+        started=motd_first_start_time - timedelta(minutes=1),
+    )
+    await resource_action.insert()
+    resource_action.add_logs([data.LogLine.log(logging.INFO, "Successfully stored version %(version)d", version=0)])
+    await resource_action.save()
+
+    action_ids_with_the_same_timestamp = []
+    for i in range(1, 6):
+        action_id = uuid.uuid4()
+        action_ids_with_the_same_timestamp.append(action_id)
+        resource_action = data.ResourceAction(
+            environment=env.id,
+            version=i,
+            resource_version_ids=[f"std::File[agent1,path=/etc/motd],v={i}"],
+            action_id=action_id,
+            action=const.ResourceAction.deploy,
+            started=motd_first_start_time,
+        )
+        await resource_action.insert()
+        resource_action.add_logs([data.LogLine.log(logging.INFO, "Successfully stored version %(version)d", version=i)])
+        await resource_action.save()
+    action_ids_with_the_same_timestamp = sorted(action_ids_with_the_same_timestamp, reverse=True)
+    action_ids_with_increasing_timestamps = []
+    for i in range(6, 11):
+        action_id = uuid.uuid4()
+        action_ids_with_increasing_timestamps.append(action_id)
+        resource_action = data.ResourceAction(
+            environment=env.id,
+            version=i,
+            resource_version_ids=[f"std::File[agent1,path=/etc/motd],v={i}"],
+            action_id=action_id,
+            action=const.ResourceAction.deploy,
+            started=motd_first_start_time + timedelta(minutes=i),
+        )
+        await resource_action.insert()
+        resource_action.add_logs([data.LogLine.log(logging.INFO, "Successfully stored version %(version)d", version=i)])
+        await resource_action.save()
+    action_ids_with_increasing_timestamps = action_ids_with_increasing_timestamps[::-1]
+    for i in range(0, 11):
+        res1 = data.Resource.new(
+            environment=env.id,
+            resource_version_id="std::File[agent1,path=/etc/motd],v=%s" % str(i),
+            status=const.ResourceState.deployed,
+            last_deploy=datetime.now() + timedelta(minutes=i),
+            attributes={"attr": [{"a": 1, "b": "c"}], "path": "/etc/motd"},
+        )
+        await res1.insert()
+
+    result = await client.get_resource_actions(
+        tid=env.id,
+        resource_type="std::File",
+        attribute="path",
+        attribute_value="/etc/motd",
+        last_timestamp=motd_first_start_time + timedelta(minutes=7),
+        limit=2,
+    )
+    assert result.code == 200
+    resource_actions = result.result["data"]
+    expected_action_ids = [action_ids_with_increasing_timestamps[-1]] + action_ids_with_the_same_timestamp[:1]
```

Comment:
I would expect that this query returns the latest two ResourceActions. This means the ResourceActions from the action_ids_with_increasing_timestamps list. Am I missing something here?

Reply:
Yes, the latest that are earlier than star + 7 minutes. There were some unused `ResourceAction`s in the test case, I removed them.